cluster/images/conformance/go-runner/run-conformance.go

```diff
@@ -0,0 +1,341 @@
+
+	// logFileName is the name of the file which stdout is tee'd to.
+	logFileName = "e2e.log"
+
+	// Misc env vars which were explicitly supported prior to the go runner.
+	dryRunEnvKey     = "E2E_DRYRUN"
+	parallelEnvKey   = "E2E_PARALLEL"
+	focusEnvKey      = "E2E_FOCUS"
+	skipEnvKey       = "E2E_SKIP"
+	providerEnvKey   = "E2E_PROVIDER"
+	kubeconfigEnvKey = "KUBECONFIG"
+	ginkgoEnvKey     = "GINKGO_BIN"
+	testBinEnvKey    = "TEST_BIN"
+
+	// extraGinkgoArgsEnvKey, if set, will is a list of other arguments to pass to ginkgo.
+	// These are passed before the test binary and include things like `--afterSuiteHook`.
+	extraGinkgoArgsEnvKey = "E2E_EXTRA_GINKGO_ARGS"
+
+	// extraArgsEnvKey, if set, will is a list of other arguments to pass to the tests.
+	// These are passed after the `--` and include things like `--provider`.
+	extraArgsEnvKey = "E2E_EXTRA_ARGS"
+
+	// extraArgsSeparaterEnvKey specifies how to split the extra args values. If unset,
+	// it will default to splitting by spaces.
+	extraArgsSeparaterEnvKey = "E2E_EXTRA_ARGS_SEP"
+
+	defaultSkip         = ""
+	defaultFocus        = "[Conformance]"
+	defaultProvider     = "local"
+	defaultParallel     = "1"
+	defaultResultsDir   = "/tmp/results"
+	defaultGinkgoBinary = "/usr/local/bin/ginkgo"
+	defaultTestBinary   = "/usr/local/bin/e2e.test"
+
+	// serialTestsRegexp is the default skip value if running in parallel. Will not
+	// override an explicit E2E_SKIP value.
+	serialTestsRegexp = "[Serial]"
+)
+
+// Getenver is the interface we use to mock out the env for easier testing. OS env
+// vars can't be as easily tested since internally it uses sync.Once.
+type Getenver interface {
+	Getenv(string) string
+}
+
+// osEnv uses the actual os.Getenv methods to lookup values.
+type osEnv struct{}
+
+func (*osEnv) Getenv(s string) string {
+	return os.Getenv(s)
+}
+
+// explicitEnv uses a map instead of os.Getenv methods to lookup values.
+type explicitEnv struct {
+	vals map[string]string
+}
+
+func (e *explicitEnv) Getenv(s string) string {
+	return e.vals[s]
+}
+
+// defaultOSEnv uses a Getenver to lookup values but if it does
+// not have that value, it falls back to its internal set of defaults.
+type defaultEnver struct {
+	firstChoice Getenver
+	defaults    map[string]string
+}
+
+func (e *defaultEnver) Getenv(s string) string {
+	v := e.firstChoice.Getenv(s)
+	if len(v) == 0 {
+		return e.defaults[s]
+	}
+	return v
+}
+
+func envWithDefaults(defaults map[string]string) Getenver {
+	return &defaultEnver{firstChoice: &osEnv{}, defaults: defaults}
+}
+
+func main() {
+	env := envWithDefaults(map[string]string{
+		resultsDirEnvKey: defaultResultsDir,
+		skipEnvKey:       defaultSkip,
+		focusEnvKey:      defaultFocus,
+		providerEnvKey:   defaultProvider,
+		parallelEnvKey:   defaultParallel,
+		ginkgoEnvKey:     defaultGinkgoBinary,
+		testBinEnvKey:    defaultTestBinary,
+	})
+
+	if err := run(env); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(env Getenver) error {
```

Comment:
So I know what you're doing, but the logic here need documentation in the code, otherwise other readers have 0 context.   Also the name run (Env) is awkward.  
It seems like you are actually doing an init(), then a run().